apptail/storage/tail_tracker.go

```diff
@@ -0,0 +1,173 @@
+package storage
+
+import (
+	"fmt"
+	"github.com/ActiveState/log"
+	"runtime"
+	"sync"
+	"sync/atomic"
+	"time"
+)
+
+type Tracker interface {
+	Update(instKey string, childKey string, childVal int64)
+	LoadTailers()
+	Remove(key string)
+	RegisterInstance(instKey string)
+	InitializeChildNode(instKey string, childkey string, offSet int64)
+	Submit()
+	StartSubmissionTimer(retentionPeriod time.Duration)
+	IsInstanceRegistered(instKey string) bool
+	IsChildNodeInitialized(instKey string, childkey string) bool
+	GetFileCachedOffset(instkey string, fname string) int64
+}
+
+type BoxedInt64 struct{ V int64 }
```

Comment:
Ah, okay, so we rely on gob to do the serialization; that means it (probably) won't attempt to do the read atomically either.  Which might be close enough for our purposes...  Still would prefer an `atomic.LoadInt64()` though.